src/libraries/System.Text.RegularExpressions/tools/GenerateRegexCasingTable.csproj

```
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
       <OutputType>Exe</OutputType>
-      <TargetFramework>net6.0</TargetFramework>
+      <TargetFramework>net7.0</TargetFramework>
```

Use the msbuild property so that this automatically moves forward.

```suggestion
      <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
```